src/net_processing.cpp

```diff
@@ -1513,6 +1553,13 @@ bool static ProcessMessage(CNode* pfrom, std::string strCommand, CDataStream& vR
             if (fReachable)
                 vAddrOk.push_back(addr);
         }
+        pfrom->m_addr_processed += num_proc;
+        pfrom->m_addr_rate_limited += num_rate_limit;
+        CNodeState* state = State(pfrom->GetId());
+        state->amt_addr_processed += num_proc;
+        state->amt_addr_rate_limited += num_rate_limit;
```

Comment:
Those 3 lines in which you create a CNodeState and update its values are useless and can be removed

Reply:
nvm I should not review at midnight lol and the line is actually needed.
But now my question becomes: why defining the same two variables both in CNode: 

std::atomic<uint64_t> m_addr_rate_limited{0};
std::atomic<uint64_t> m_addr_processed{0};

and then in CNodeState

uint64_t amt_addr_processed = 0;
uint64_t amt_addr_rate_limited = 0;

Since the former is not even used I'd just suggest removing completely the one defined in CNode

Reply:
Resolved